Framework/PythonInterface/plugins/algorithms/ResetNegatives2D.py

```diff
@@ -0,0 +1,65 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
```

Comment:
Just have some general comments on the algorithm, but also a query:

Would the following do exactly the same as what the algorithm is doing?

```
intMin = np.min(data.extractY())
# Check if minimal Intensity is negative. If it is, add -1*intMin to all intensities
if intMin < 0:
    Scale(InputWorkspace=data, OutputWorkspace=data, Factor=-intMin, Operation="Add")
```
Maybe I'm missing an edge case, as this is not my area of expertise. But extractY() returns a 2D view of the workspace, giving you the zData for each y and x bin. np.min would return the minimum value in the 2D data, which we can then shift everything up by using Scale.

Reply:
That actually works very nicely, at least in initial testing with simulated data. I kind of feel stupid because I didnt think of this 😄 I will do some more tests with actual data to make sure its the same.